src/coreclr/src/gc/gc.cpp

```diff
@@ -15510,6 +15512,7 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
                                            BOOL* blocking_collection_p
                                            STRESS_HEAP_ARG(int n_original))
 {
+    gc_heap::gc_data_global.gen_to_condemn_reasons.init();
```

Comment:
we are introducing a new scenario here, which is we are  no longer only fill in the global data at the end of a GC. this means we'll need to save a copy for BGC. see comments for this line:

`memcpy (&bgc_data_per_heap, &gc_data_per_heap, sizeof(gc_data_per_heap));`